test/e2e/reclaim.go

```diff
@@ -6,18 +6,207 @@ import (
+			},
+		})
+
+		defer cleanupTestContext(ctx)
+
+		By("Setup initial jobs")
+
+		_, err := CreateReclaimJob(ctx, CPU1Mem1, j1, q1, "")
+		Expect(err).NotTo(HaveOccurred(), "Wait for job1 failed")
+
+		_, err = CreateReclaimJob(ctx, CPU1Mem1, j2, q2, "")
+		Expect(err).NotTo(HaveOccurred(), "Wait for job2 failed")
+
+		By("Create new comming queue and job")
+		q3 := "reclaim-q3"
+		ctx.queues = append(ctx.queues, q3)
+		createQueues(ctx)
+
+		err = WaitQueueStatus(ctx, "Open", 1, q1)
+		Expect(err).NotTo(HaveOccurred(), "Error waiting for queue open")
+
+		_, err = CreateReclaimJob(ctx, CPU1Mem1, j3, q3, "")
+		Expect(err).NotTo(HaveOccurred(), "Wait for job3 failed")
+
+		// delete pod of job3 to make sure reclaim-j3 podgroup is pending
+		listOptions := metav1.ListOptions{
+			LabelSelector: labels.Set(map[string]string{"volcano.sh/job-name": j3}).String(),
+		}
+
+		job3pods, err := ctx.kubeclient.CoreV1().Pods(ctx.namespace).List(context.TODO(), listOptions)
+		Expect(err).NotTo(HaveOccurred(), "Get %s pod failed", j3)
+		for _, pod := range job3pods.Items {
+			err = ctx.kubeclient.CoreV1().Pods(pod.Namespace).Delete(context.TODO(), pod.Name, metav1.DeleteOptions{})
+			Expect(err).NotTo(HaveOccurred(), "Failed to delete pod %s", pod.Name)
+		}
+
+		By("Make sure all job running")
+
+		err = WaitQueueStatus(ctx, "Running", 1, q1)
+		Expect(err).NotTo(HaveOccurred(), "Error waiting for queue running")
+
+		err = WaitQueueStatus(ctx, "Running", 2, q2)
+		Expect(err).NotTo(HaveOccurred(), "Error waiting for queue running")
+
+		err = WaitQueueStatus(ctx, "Pending", 1, q3)
+		Expect(err).NotTo(HaveOccurred(), "Error waiting for queue pending")
+	})
+
+	It("Reclaim Cases 4: New queue with job created no reclaim when new queue is not created", func() {
+		q1 := "default"
+		q2 := "reclaim-q2"
+		ctx := initTestContext(options{
+			queues:             []string{q2},
+			nodesNumLimit:      3,
+			nodesResourceLimit: CPU1Mem1,
+			priorityClasses: map[string]int32{
+				"low-priority":  10,
+				"high-priority": 10000,
+			},
+		})
+
+		defer cleanupTestContext(ctx)
+
+		By("Setup initial jobs")
+
+		_, err := CreateReclaimJob(ctx, CPU1Mem1, "reclaim-j1", q1, "")
+		Expect(err).NotTo(HaveOccurred(), "Wait for job1 failed")
+
+		_, err = CreateReclaimJob(ctx, CPU1Mem1, "reclaim-j2", q2, "")
+		Expect(err).NotTo(HaveOccurred(), "Wait for job2 failed")
+
+		By("Create new comming job")
+		q3 := "reclaim-q3"
+
+		_, err = CreateReclaimJob(ctx, CPU1Mem1, "reclaim-j3", q3, "")
+		Expect(err).Should(HaveOccurred(), "job3 create failed when queue3 is not created")
+
+		By("Make sure all job running")
+
+		err = WaitQueueStatus(ctx, "Running", 1, q1)
+		Expect(err).NotTo(HaveOccurred(), "Error waiting for queue running")
+
+		err = WaitQueueStatus(ctx, "Running", 1, q2)
+		Expect(err).NotTo(HaveOccurred(), "Error waiting for queue running")
+	})
+
+	It("Reclaim Cases 5: New queue with job created no reclaim when job or task is low-priority", func() {
```

Comment:
Add a comment: as we agreed, this is not intended behavior. Actually, it is a bug.